sstables/sortedmap_key_index.go

```diff
@@ -0,0 +1,173 @@
+package sstables
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	rProto "github.com/thomasjungblut/go-sstables/recordio/proto"
+	"github.com/thomasjungblut/go-sstables/skiplist"
+	"github.com/thomasjungblut/go-sstables/sstables/proto"
+)
+
+// SortedMapIndex is keeping the entire index as a map and the ordered keys as a slice in memory
+// Use binary search to iter over the index.
+type ByteKeyMapper[T comparable] interface {
+	MapBytes(data []byte) T
+}
+
+// use this key Mapper for sha1 keys
+type Byte20KeyMapper[T comparable] struct{}
+
+func (m Byte20KeyMapper[T]) MapBytes(data []byte) T {
+	var result [20]byte
+
+	// Gestion de la longueur
+	if len(data) > 20 {
+		data = data[:20]
+	}
+
+	// Copie des données
+	copy(result[:], data)
+
+	// Conversion générique - nécessite que T soit [20]byte
+	return any(result).(T)
+}
+
+type SortedMapIndex[T comparable] struct {
+	sliceKeyIndex SliceKeyIndex
```

Comment:
if you directly embed without the identifier so just:

```
type SortedMapIndex[T comparable] struct {
	SliceKeyIndex
        ... 
```

you can also skip implementing these proxy iterator functions below

Reply:
nice